sklearn_crfsuite/estimator.py

```diff
@@ -307,7 +307,7 @@ def fit(self, X, y, X_dev=None, y_dev=None):
         trainer = self._get_trainer()
         train_data = zip(X, y)
 
-        if self.verbose:
+        if self.verbose and "len" in dir(X):
```

Comment:
This looks incorrect:

```py
>>> "len" in dir([])
False
```

Actually I'm not sure why we're passing len(X) here, tqdm should know how to get it. Probably it was a workaround for some issue, but maybe it can be removed now.

The same issue is present for X_dev.

Reply:
Oops , will close this PR , will give a new one .